frontend/src/home/SwitchPlanPopper.tsx

```diff
@@ -99,17 +157,48 @@ export class SwitchPlanPopperComponent extends React.Component<
     });
   }
 
+  handleDelete(planId: number, name: string) {
+    if (this.props.schedules.length < 2) {
+      this.setState({
+        errorSnackbarOpen: true,
+      });
+    } else if (this.props.userId && this.props.token) {
+      deletePlanForUser(this.props.userId, planId, this.props.token);
+      this.props.deletePlan(name);
+      this.props.deletePlanIdFromUserState(planId);
+    }
+  }
+
+  handleCloseErrorSnackbar() {
+    this.setState({
+      errorSnackbarOpen: false,
+    });
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.activeSchedule.name !== prevProps.activeSchedule.name) {
```

Comment:
Since the activeSchedule sometimes needs to load in first, add a check here to make sure activeSchedule isn't undefined (causes an error when the user first logs in)